### PR TITLE
Adds commandline tool to list filenames for vectorization and optionally save to a file

### PR DESCRIPTION
**Pull request by danielhertenstein**
Addresses #97 

This PR adds a command line tool to aid in vectorization. You can list the files in a directory for easy copy-pasting into the vectorizer box in Fathom-Fox. Optionally, you can save the output to a file so you can reference the file later and forgo using this tool every time.

### CODE REVIEW

File: cli/fathom_web/list.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from click import argument, command, File, option, Path
+import pathlib
+
+
+@command()
+@option('--out_file', '-o', type=File(mode='w'), default=None,
```
Comment (erikrose) [inline cli/fathom_web/list.py]: If we change this to `--out-file` (with a hyphen), we'll be consistent with fathom-train and the other tools. `click` will change it to an underscore when stuffing its value in a variable.

File: cli/fathom_web/list.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        help='A file for saving the printed filenames for easy future reference.')
+@argument('in_directory', type=Path(exists=True, file_okay=False, allow_dash=True))
+def main(in_directory, out_file):
+    """
+    Lists filenames in a IN_DIRECTORY, one filename per line. Optionally saves output to OUT_FILE.
+
+    Ignores hidden files.
+
+    This is useful for vectorizing samples using fathom-fox. Fathom-fox expects input filenames copied into a text box
```
Comment (erikrose) [inline cli/fathom_web/list.py]: FathomFox

File: cli/fathom_web/list.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        if file.name.startswith('.'):
+            continue
+
+        print(file.name)
+
+        if out_file is not None:
+            filenames_to_save.append(file.name + '\n')
+
+    if out_file is not None:
+        out_file.writelines(filenames_to_save)
```
Comment (erikrose) [inline cli/fathom_web/list.py]: Where ever did you find this `writelines` routine? click doesn't document it, and the only one I can find in Python 3 is in asyncio. Its docs say we should be calling drain() afterward. Should this be a higher-level routine, or should I be enlightened?

File: cli/fathom_web/list.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+from click import argument, command, File, option, Path
+import pathlib
+
+
+@command()
+@option('--out_file', '-o', type=File(mode='w'), default=None,
+        help='A file for saving the printed filenames for easy future reference.')
+@argument('in_directory', type=Path(exists=True, file_okay=False, allow_dash=True))
```
Comment (erikrose) [inline cli/fathom_web/list.py]: I'm not sure that allow_dash is applicable for a dir.

File: cli/fathom_web/list.py (hunk 3 above)
Comment (erikrose) [reply cli/fathom_web/list.py]: I am enlightened!

File: cli/fathom_web/list.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
+from click import argument, command, File, option, Path
+import pathlib
```
[the remark this answers is not in the document]
Comment (danielhertenstein) [reply cli/fathom_web/list.py]: Will do!

File: cli/fathom_web/list.py (hunk 1 above)
Comment (danielhertenstein) [reply cli/fathom_web/list.py]: Will do!

File: cli/fathom_web/list.py (hunk 2 above)
Comment (danielhertenstein) [reply cli/fathom_web/list.py]: Whoops. I went by the repo name instead of checking the official name.

File: cli/fathom_web/list.py (hunk 4 above)
Comment (danielhertenstein) [reply cli/fathom_web/list.py]: As we spoke about, I will remove it.

**Pull request merged**